Problem with FI_ADDR_NOTAVAIL in C++

When trying to compile a C++ application that includes the use of `FI_ADDR_NOTAVAIL`, one gets this kind of compile error:

```
my_program.cc:249: error: ‘UINT64_MAX’ was not declared in this scope
```

It looks like `FI_ADDR_NOTAVAIL` (and `FI_ADDR_UNSPEC`) is defined in `fabric.h` as:

```c
#define FI_ADDR_UNSPEC          UINT64_MAX
#define FI_ADDR_NOTAVAIL        UINT64_MAX
```

According to a comment in `<stdint.h>` right above where `UINT64_MAX` is defined:

```c
/* The ISO C99 standard specifies that in C++ implementations these         
   macros should only be defined if explicitly requested.  */
```

I don't think we should force C++ programs to request that these macros be defined.  I'm not a language expert to know what the Right way is to do this.  Here's one suggestion -- is this acceptable?

```c
#define FI_ADDR_UNSPEC     ((uint64_t) ~0)
#define FI_ADDR_NOTAVAIL   ((uin64_t) ~0)
```

***EDIT:*** Added `(uint64_t)` cast.

I'm happy to make a PR with whatever the Right solution is deemed to be.

Arrgh.  g++ v5.1 and v5.2 (the only versions I tested) balk at trying to include `<cstdint>`:

```
$ g++ foo.c
In file included from /usr/local/include/c++/5.1.0/cstdint:35:0,
                 from foo.c:4:
/usr/local/include/c++/5.1.0/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
```

Soooo... I'm not going to try to solve that one today.  I'll just change the defines of `FI_ADDR_[UNSPEC|NOTAVAIL]`.

--- a/include/rdma/fabric.h
+++ b/include/rdma/fabric.h
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2013-2016 Intel Corporation. All rights reserved.
+ * Copyright (c) 2016 Cisco Systems, Inc. All rights reserved.
  *
  * This software is available to you under a choice of one of two
  * licenses.  You may choose to be licensed under the terms of the GNU
@@ -157,8 +158,8 @@ enum {
 	FI_ADDR_GNI
 };
 
-#define FI_ADDR_UNSPEC		UINT64_MAX
-#define FI_ADDR_NOTAVAIL	UINT64_MAX
+#define FI_ADDR_UNSPEC		((uint64_t) -1)
+#define FI_ADDR_NOTAVAIL	((uint64_t) -1)
 #define FI_SHARED_CONTEXT	(-(size_t)1)
 typedef uint64_t		fi_addr_t;
 

